wikimedia/wikipedia-ios
# Talk pages - non-language-specific project tweaks

**Phabricator:** 
N/A

### Notes
These are a couple of tweaks that jumped out at me since integrating `WikimediaProject` (for the header work) and our caching for offline mode.

e43b552 - This is a follow-on from [this feedback](https://github.com/wikimedia/wikipedia-ios/pull/4340#discussion_r989122763) to improve the cache file names for projects that do not have languages associated. I originally thought I'd lean on WikimediaProject (which hadn't been merged yet) for this feedback, but languageVariantCode seems to do what I wanted just fine. This removes the unnecessary subdomain that was showing for non-language-specific project file names:

Before: (note commons, wikidata, and test all have an unnecessary extra subdomain):
![Screen Shot 2022-10-14 at 10 45 28 AM](https://user-images.githubusercontent.com/3620196/195899013-cc021e36-dc95-489c-a42e-84ae270b7b21.png)

After (ZH uses variant code as part of file name, to keep it unique from other ZH variants):
![Screen Shot 2022-10-14 at 10 49 31 AM](https://user-images.githubusercontent.com/3620196/195899137-f5386c90-257c-4715-abc4-f6df5edf4f64.png)

908a8b5 - This hides the "Change Language" option in the overflow menu for talk pages that are not language-specific (like Commons or Wikidata).

### Test Steps
1. Trigger a notification to your talk page on Commons or Wikidata. Visit the talk page from Notifications Center. Enter airplane mode. Revisit the page. Confirm offline mode works and you still see data.
2. Turn off airplane mode. Confirm Commons and Wikidata talk pages do not show "Change language" option in the overflow menu. Confirm Wikipedia talk pages still show "Change language" option in overflow menu.
